src/components/ContentSidebar/ContentSidebar.js

```diff
@@ -468,6 +475,53 @@ class ContentSidebar extends PureComponent<Props, State> {
      */
     fetchTasksSuccessCallback = (tasks: Tasks): void => {
         this.setState({ tasks, tasksError: undefined });
+        this.fetchTaskAssignments(tasks);
+    };
+
+    /**
+     * Update Tasks to include task assignments
+     *
+     * @private
+     * @param {Array<Task>} entries - Box task entries
+     * @param {TaskAssignments} assignments - Box task assigments
+     * @return {TaskAssignments} Updated Box task assignments
+     */
+    replaceTaskAssignments(entries, assignments) {
+        return {
+            total_count: assignments.entries.length,
+            entries: entries.map((item) => {
+                const assignment = assignments.entries.find((a) => a.id === item.id);
+                if (assignment) {
+                    return {
+                        ...assignment
+                    };
+                }
+                return item;
+            })
+        };
+    }
+
+    /**
+     * File task assignment fetch success callback
+     *
+     * @private
+     * @param {TaskAssignments} assignments - Box task assigment
+     * @return {void}
+     */
+    fetchTaskAssignmentsSuccessCallback = (assignments: TaskAssignments): void => {
+        const { entries, total_count } = this.state.tasks;
+        this.setState({
+            tasksWithAssignments: {
```

Comment:
We cannot update the tasks state at any point because that would cause a render(). We need to maintain an internal state and then only set the tasks object in the state when all the assignments have been fetched